taxcalc/functions.py

```diff
@@ -1604,60 +1583,38 @@ def C1040(e07400, e07180, e07200, c07220, c07230, e07250,
 
 
 @iterate_jit(nopython=True)
-def DEITC(c08795, c59660, c09200, c07100, c08800, c05800, _othertax):
+def DEITC(c08795, c59660, c09200, c07100, c08800, c05800, _avail, _othertax):
 
     # Decomposition of EITC
-
-    if c08795 > 0 and c59660 > 0 and c08795 <= c59660:
-        c59680 = c08795
-
-        _comb = c59660 - c59680
-
-    else:
-
-        c59680 = 0.
-        _comb = 0.
-
-    if c08795 > 0 and c59660 > 0 and c08795 > c59660:
-        c59680 = c59660
+    _comb = 0
+    c10950 = 0
+    c59680 = min(c59660, max(0, _avail))
+    _avail = _avail - c59680
+    _avail = _avail + _othertax
+    c59700 = min(_avail, c59660 - c59680)
+    c59720 = c59660 - c59680 - c59700
 
     c07150 = c07100 + c59680
     c07150 = min(c07150, c05800)
     c08800 = c05800 - c07150
 
-    c59700 = min(_othertax, c59680)
-    c59720 = c59680 - c59700
-
-    if c08795 < 0 or c59660 <= 0:
-        _compb = 0.
-        c59680 = 0.
-        c59700 = 0.
-        c59720 = 0.
-
-    else:
-        _compb = 0.
-
-    c07150 = c07100 + c59680
-    c07150 = c07150
-    c10950 = 0.
-
     return (c59680, c59700, c59720, _comb, c07150, c10950, c08800)
 
 
 @iterate_jit(nopython=True)
 def OSPC_TAX(c09200, c59660, c11070, c10960, c11600, c10950, _eitc, e11580,
              e11450, e11500, e82040, e09900, e11400, e11300, e11200, e11100,
-             e11550, e09710, e09720):
+             e11550, e09710, e09720, e10000):
 
-    _refund = (c59660 + c11070 + c11600 + c10960 + c10950 + e11580 + e11450 +
+    _refund = (c59660 + c11070 + c10960 + c10950 + c11600 + e11580 + e11450 +
                e11500)
 
     _ospctax = c09200 - _refund - e82040
 
-    _payments = (e09900 + e11500 + e11400 + e11300 + e11200 + e11100 + e11550 +
+    _payments = (c59660 + c10950 + c10960 + c11070 + e10000 + e11550 + e11580 +
                  e11450)
 
-    c10300 = max(0, _ospctax - e09710 - e09720 - _payments)
+    c10300 = max(0, c09200 - _payments)
```

Comment:
Calculation of `c10300` got changed significantly. Essentially, `c11600`, `e09900`, `e11100`, `e11200`, `e11300`, `e11400` and `e11500` are not subtracted from `c09200` anymore. most of the updates look good to, expect two parts:
1. `e11300` corresponds to line 72 in form 1040, and thus is part of the payments. Should we include it in the calculation of `_payments`?
2. Although `e10000` takes zero value, not sure if we should include it in `_payments`. 
 It's also interesting to note that, in old version, `e11450` and `e11500`, although take zero values, were being deducted twice from `c09200`, which doesn't seem reasonable.